SIL.Core/IO/FileHelper.cs

```diff
@@ -40,5 +42,43 @@ public static bool Grep(string inputPath, string pattern)
 			}
 			return false;
 		}
+
+		/// <summary>
+		/// Serialize an object directly to disk, avoiding the operating system cache.
+		/// 
+		/// This method is intended to prevent the problem with null files being generated
+		/// as reported here: LT-20651, LT-20333, LTB-3915, LTB-3916, and LTB-3917
+		/// The theory is that FieldWorks is closing and power is lost before the operating
+		/// system cache is written to disk. This method is intended to prevent that problem
+		/// by avoiding the cache and writing directly to disk.
+		///
+		/// The disadvantage of this method is that it is slow; so it should only be used to
+		/// write relatively small files that are not frequently written.
+		/// </summary>
+		/// <param name="objToSerialize">Object to be serialized to the file.</param>
+		/// <param name="path">The full path (containing the file name and extension).</param>
+		public static void WriteXmlFileDirectlyToDisk(object objToSerialize, string path)
```

Comment:
RobustFile already has a Create method that creates a FileStream with WriteThrough. I wonder if you can just use that or make this method use it. In any case, it kind of seems that this logic maybe belongs in that class rather than in FileHelper.

Reply:
Alternatively, could this be put as another overload on XmlSerializationHelper.SerializeToFile? That would probably make it more easily discoverable.

Reply:
Thanks Tom, I wasn't aware of these classes.  I'll take a look and see where it seems to fit best.

Reply:
@tombogle I moved the utility to XmlSerializationHelper and changed it to call RobustFile.Create().